addons/binding/org.openhab.binding.mcp23017/src/main/java/org/openhab/binding/mcp23017/handler/Mcp23017Handler.java

```diff
@@ -98,12 +98,29 @@ private boolean verifyChannel(ChannelUID channelUID) {
 
     private void handleOutputCommand(ChannelUID channelUID, Command command) {
         GpioPinDigitalOutput outputPin = pinStateHolder.getOutputPin(channelUID);
-        logger.debug("got output pin {} for channel {} and command {}", outputPin, channelUID, command);
-        if (OnOffType.ON == command) {
-            GPIODataHolder.GPIO.setState(PinState.HIGH, outputPin);
-        } else if (OnOffType.OFF == command) {
-            GPIODataHolder.GPIO.setState(PinState.LOW, outputPin);
+        
+        Configuration configuration = this.getThing().getChannel(channelUID.getId()).getConfiguration();
+        String invertLogic = (String) configuration.get(ACTIVE_LOW);        
+        PinState newState = null;
+        
+        // invertLogic is null if not configured
+        boolean activeLowFlag = ACTIVE_LOW_ENABLED.equalsIgnoreCase(invertLogic);
+        if (activeLowFlag) {
```

Comment:
I think you could make the code much simpler. Something like this:
```
if (command instanceof OnOffType) {
    Configuration configuration = this.getThing().getChannel(channelUID.getId()).getConfiguration();
    boolean inverseLogic = Boolean.parseBoolean(configuration.get(INVERSE).toString());
    PinState pinState = command == OnOffType.ON ^ inverseLogic ? PinState.HIGH : PinState.LOW;
    GPIODataHolder.GPIO.setState(pinState, outputPin);
}
```

**Your current code also contains a bug**. If the command is not an `OnOffType` you'll be passing the value `null` as `newState`. That is not how it worked before.

Reply:
Good finding! Thank you! Ill improve my code!

Reply:
> Good finding! Thank you! Ill improve my code!

@igorarenz did you already do this?

Reply:
Its commited now.